File: src/main/java/kitchenpos/menus/infra/DefaultProductClient.java
package kitchenpos.menus.infra;

import kitchenpos.menus.domain.tobe.menu.ProductClient;
import kitchenpos.products.domain.tobe.ProductRepository;
import org.springframework.stereotype.Component;

import java.util.List;
import java.util.UUID;
@Component
public class DefaultProductClient implements ProductClient {
  private final ProductRepository productRepository;

  public DefaultProductClient(final ProductRepository productRepository) {
    this.productRepository = productRepository;
  }

  @Override
  public int countProductIds(final List<UUID> productIds) {
    return productRepository.findAllByIdIn(productIds).size();
  }
}

File: src/main/kotlin/kitchenpos/menu/tobe/infra/DefaultProductClient.kt
package kitchenpos.menu.tobe.infra

import java.util.*
import kitchenpos.menu.tobe.domain.ProductClient
import kitchenpos.menu.tobe.domain.ProductInfo
import kitchenpos.product.tobe.domain.ProductRepository
import org.springframework.stereotype.Service

@Service
class DefaultProductClient(
    private val productRepository: ProductRepository,
) : ProductClient {
    override fun getProducts(productIds: List<UUID>): Map<UUID, ProductInfo> {
        return productRepository.findAllByIdIn(productIds)
            .associate { it.id to ProductInfo(it.id, it.productPrice.price) }
    }
}
